imports/plugins/core/cart/server/no-meteor/startup.js

```diff
@@ -29,47 +30,60 @@ function getFlatVariantsAndOptions(catalogProductVariants) {
  * @returns {Promise<null>} Promise that resolves with null
  */
 async function updateAllCartsForVariant({ Cart, context, variant }) {
-  const { appEvents, queries } = context;
+  const { mutations, queries } = context;
   const { variantId } = variant;
 
+  Logger.debug({ ...logCtx, variantId, fn: "updateAllCartsForVariant" }, "Running updateAllCartsForVariant");
+
+  let updatedCarts = [];
+
+  /**
+   * @summary Bulk save an array of updated carts
+   * @return {undefined}
+   */
+  async function saveCarts() {
+    if (updatedCarts.length === 0) return;
+    await mutations.saveManyCarts(context, updatedCarts);
+    updatedCarts = [];
+  }
+
+  /**
+   * @summary Get updated prices for a single cart, and check whether there are any changes.
```

Comment:
I think we should be adding/setting a flag here for when the cart has been modified so that the UX could display some sort of message.